pkg/stores/sqlproxy/proxy_store.go

```diff
@@ -53,12 +53,59 @@ var (
 	paramScheme               = runtime.NewScheme()
 	paramCodec                = runtime.NewParameterCodec(paramScheme)
 	typeSpecificIndexedFields = map[string][][]string{
-		"_v1_Namespace": {{`metadata`, `labels[field.cattle.io/projectId]`}},
-		"_v1_Node":      {{`status`, `nodeInfo`, `kubeletVersion`}, {`status`, `nodeInfo`, `operatingSystem`}},
-		"_v1_Pod":       {{`spec`, `containers`, `image`}, {`spec`, `nodeName`}},
-		"_v1_ConfigMap": {{`metadata`, `labels[harvesterhci.io/cloud-init-template]`}},
-
-		"management.cattle.io_v3_Node": {{`status`, `nodeName`}},
+		gvkKey("", "v1", "Event"): {
+			{"_type"},
```

Comment:
The syntax of this type make me think that this is a reserved field. Basically, Steve has a few reserved fields which it adds to an object (id is another). When an object already has this field, Steve moves the original field to "_fieldName".

Vai doesn't store the modified versions of these objects - instead, it stores the original. You'll need to use the new virtual fields logic (take a look [here](https://github.com/rancher/steve/blob/main/pkg/resources/virtual/virtual.go)) to start to make sure that this is filterable/sortable.

You can also try testing this to see if I am wrong.

Reply:
This converts to key `"_v1_Event"`, I don't see that looks like a reserved name.

Also this block is a duplicate of `gvkKey("events.k8s.io", "v1", "Event")` and it's possible that we don't need this block.

Reply:
Sorry for not clarifying - it's not the resource name that is the problem, it's the `_type` field.

Reply:
Testing shows that for `/v1/events`, specifying `?filter=type=STRING` converts `type` into `_type`:
```
$ curl -skL ... /v1/events' | jq . | egrep '"_?type"'
  "type": "collection",
      "type": "event",
      "_type": "Normal",
      "type": "event",
      "_type": "Warning",
…
# Searching for 'type=event' gives the empty-list returned JSON payload:
$ curl -skL ... /v1/events'?filter=type=event' | wc
       1       1     197
# Searching for `?filter=type=....` converts `type` to `_type`
$ curl -skL ... /v1/events'?filter=type=Normal' | wc
       1       2981  602473

Reply:
As for `events`, when I look at the dashboard PR it looks like they're matching on `_v1_Event` and not `events.k8s.io_v1_Event`

Reply:
The filter needs to apply on _type - since that's the field that they see they need to be able to filter on the final field rather than on the original field before steve's transformation. Please give this another look.